nilearn/connectome/tests/test_connectivity_matrices.py

```diff
@@ -453,104 +518,140 @@ def test_connectivity_measure_errors():
+)
+def test_connectivity_measure_tangent(cov_estimator, signals_and_covariances):
+    signals, covs = signals_and_covariances
+    kind = "tangent"
+
+    conn_measure = ConnectivityMeasure(kind=kind, cov_estimator=cov_estimator)
+    connectivities = conn_measure.fit_transform(signals)
+
+    for k, cov_new in enumerate(connectivities):
+        # Positive definiteness if expected and output value checks
+
+        assert_array_almost_equal(cov_new, cov_new.T)
+        gmean_sqrt = _map_eigenvalues(np.sqrt, conn_measure.mean_)
+        assert is_spd(gmean_sqrt, decimal=7)
+        assert is_spd(conn_measure.whitening_, decimal=7)
+        assert_array_almost_equal(
+            conn_measure.whitening_.dot(gmean_sqrt),
+            np.eye(N_FEATURES),
         )
-        signals.append(signal)
-        signal -= signal.mean(axis=0)
-        emp_covs.append((signal.T).dot(signal) / n_samples)
-        ledoit_covs.append(ledoit_estimator.fit(signal).covariance_)
+        assert_array_almost_equal(
+            gmean_sqrt.dot(_map_eigenvalues(np.exp, cov_new)).dot(gmean_sqrt),
+            covs[k],
+        )
+
+
+@pytest.mark.parametrize(
+    "cov_estimator", [EmpiricalCovariance(), LedoitWolf()]
+)
+def test_connectivity_measure_precision(
+    cov_estimator, signals_and_covariances
+):
+    signals, covs = signals_and_covariances
+    kind = "precision"
 
-    kinds = [
+    conn_measure = ConnectivityMeasure(kind=kind, cov_estimator=cov_estimator)
+    connectivities = conn_measure.fit_transform(signals)
+
+    for k, cov_new in enumerate(connectivities):
+        assert is_spd(cov_new, decimal=7)
+        assert_array_almost_equal(cov_new.dot(covs[k]), np.eye(N_FEATURES))
+
+
+@pytest.mark.parametrize(
+    "cov_estimator", [EmpiricalCovariance(), LedoitWolf()]
+)
+def test_connectivity_measure_correlation(
+    cov_estimator, signals_and_covariances
+):
+    signals, covs = signals_and_covariances
+    kind = "correlation"
+
+    conn_measure = ConnectivityMeasure(kind=kind, cov_estimator=cov_estimator)
+    connectivities = conn_measure.fit_transform(signals)
+
+    for k, cov_new in enumerate(connectivities):
+        assert is_spd(cov_new, decimal=7)
+        d = np.sqrt(np.diag(np.diag(covs[k])))
+        if cov_estimator == EmpiricalCovariance():
+            assert_array_almost_equal(d.dot(cov_new).dot(d), covs[k])
+        assert_array_almost_equal(np.diag(cov_new), np.ones(N_FEATURES))
+
+
+@pytest.mark.parametrize(
+    "cov_estimator", [EmpiricalCovariance(), LedoitWolf()]
+)
+def test_connectivity_measure_partial_correlation(
+    cov_estimator, signals_and_covariances
+):
+    signals, covs = signals_and_covariances
+    kind = "partial correlation"
+
+    conn_measure = ConnectivityMeasure(kind=kind, cov_estimator=cov_estimator)
+    connectivities = conn_measure.fit_transform(signals)
+
+    for k, cov_new in enumerate(connectivities):
+        prec = linalg.inv(covs[k])
```

Comment:
I think we could also parametrize these tests on "kind". It seems like only the assertion part is changing, the rest is the same. So maybe something like this (let me know what you think, it's only a suggestion, I haven't tested whether this works or will end up more readable....):

```python
@pytest.mark.parametrize("cov_estimator", [EmpiricalCovariance(), LedoitWolf()])
@pytest.mark.parametrize("kind", ["covariance", "correlation", "tangent", "precision", "partial correlation"])
def test_connectivity_measure(cov_estimator, kind, signals_and_covariance):
    signals, covs = signals_and_covariances
    conn_measure = ConnectivityMeasure(kind=kind, cov_estimator=cov_estimator)
    connectivities = conn_measure.fit_transform(signals)
    connections = (connectivities, conn_measure, covs)
    _assert_connectivities(connections, kind)

def _assert_connectivities(connections: tuple, kind: str) -> bool:
    if kind == "tangent":
        return _assert_tangent_connectivities(connections)
    if kind == "correlation":
        return _assert_correlation_connectivities(connections)
    ...

def _assert_tangent_connectivities(connections: tuple) -> bool:
    """Check positive definiteness if expected."""
    connectivities, covs, conn_measure = connections
    for k, cov_new in enumerate(connectivities):
        assert_array_almost_equal(cov_new, cov_new.T)
        gmean_sqrt = _map_eigenvalues(np.sqrt, conn_measure.mean_)
        assert is_spd(gmean_sqrt, decimal=7)
        assert is_spd(conn_measure.whitening_, decimal=7)
        assert_array_almost_equal(conn_measure.whitening_.dot(gmean_sqrt), np.eye(N_FEATURES))
        assert_array_almost_equal(
             gmean_sqrt.dot(_map_eigenvalues(np.exp, cov_new)).dot(gmean_sqrt), covs[k],
         )
    return True

.... 
```

Reply:
done